src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -893,15 +1022,23 @@ private bool TryCreateSegmentForOperation(ODataPathSegment previousSegment, stri
         /// <param name="text">The text for the next segment.</param>
         private void CreateNextSegment(string text)
         {
-            // For Non-KeyAsSegment, try to handle it as a key property value, unless it was preceeded by an excape-marker segmetn ('$').
-            // For KeyAsSegment, the following precedence rules should be supported [ODATA-799]:
-            // Try to match an OData segment (starting with �$�).
-            // Try to match an alias - qualified bound action name, bound function overload, or type name.
-            // Try to match a namespace-qualified bound action name, bound function overload, or type name.
-            // Try to match an unqualified bound action name, bound function overload, or type name in a default namespace.
-            // Treat as a key.
+            string identifier;
+            string parenthesisExpression;
+            ExtractSegmentIdentifierAndParenthesisExpression(text, out identifier, out parenthesisExpression);
+
+            /*
+             * For Non-KeyAsSegment, try to handle it as a key property value, unless it was preceeded by an excape - marker segment('$').
+             * For KeyAsSegment, the following precedence rules should be supported[ODATA - 799]:
+             * Try to match an OData segment(starting with �$�).
+             *   - Note: $filter path segment is a special case that has the format "$filter=@a", where @a represents an alias.
+             * Try to match an alias - qualified bound action name, bound function overload, or type name.
+             * Try to match a namespace-qualified bound action name, bound function overload, or type name.
+             * Try to match an unqualified bound action name, bound function overload, or type name in a default namespace.
+             * Treat as a key.
+             */
+
             // $value
-            if (this.TryCreateValueSegment(text))

```

Comment:
>TryCreateValueSegment [](start = 21, length = 21)

maybe keep it unchanged to use "text"?

Reply:
We had a discussion during a design meeting (I think you were away at the time). I updated this to use `identifier` and `parenthesisExpression` because in the previous implementation, each of the segment parsers were calling `ExtractSegmentIdentifierAndParenthesisExpression` which was redundant work. I moved the `ExtractSegmentIdentifierAndParenthesisExpression` out of those methods and to line 1027 and then I pass the information into the `TryCreate...` methods.